### PR TITLE
For #216: DeckBuilder: Decklist displays current hero deck

### LINKED ISSUE #216: DeckBuilder: Decklist displays current hero deck
When in the Deck Builder screen, the currently selected Hero's decklist should be linked to the DeckList UI elements.

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #216: DeckBuilder: Decklist displays current hero deck

Created DeckBuilderState, DeckVisualizer, CardListItem prefab and wired them into DeckBuilderScene

### CODE REVIEW

File: src/ProjectNeon/Assets/Features/Cards/CardPresenter.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public void Set(Card card, Action onClick)
     {
         _onClick = onClick;
         _card = card;
-        name.text = _card.name.WithSpaceBetweenWords();
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Cards/CardPresenter.cs]: Why this change? This makes the card less readable, no?

File: src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckBuilderState.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/**
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckBuilderState.cs]: Don't include redundant comments. 

If there are any comments, then should only be explaining unique implementations.

File: src/ProjectNeon/Assets/Features/Cards/CardPresenter.cs (hunk 1 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Cards/CardPresenter.cs]: @SilasReinagel Sorry, debug code left

File: src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckBuilderState.cs (hunk 2 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckBuilderState.cs]: @SilasReinagel Ok, I'll remove them
